src/coreclr/vm/interoplibinterface_comwrappers.h

```diff
@@ -0,0 +1,175 @@
+extern "C" void QCALLTYPE ComWrappers_GetIUnknownImpl(
+    _Out_ void** fpQueryInterface,
+    _Out_ void** fpAddRef,
+    _Out_ void** fpRelease);
+
+extern "C" void* QCALLTYPE ComWrappers_AllocateRefCountedHandle(_In_ QCall::ObjectHandleOnStack obj);
+
+extern "C" void const* QCALLTYPE ComWrappers_GetIReferenceTrackerTargetVftbl();
+
+extern "C" void const* QCALLTYPE ComWrappers_GetTaggedImpl();
+
+extern "C" void QCALLTYPE ComWrappers_RegisterManagedObjectWrapperForDiagnostics(
+    _In_ QCall::ObjectHandleOnStack obj,
+    _In_ void* wrapper);
+
+extern "C" void QCALLTYPE ComWrappers_RegisterNativeObjectWrapperForDiagnostics(
+    _In_ QCall::ObjectHandleOnStack target,
+    _In_ QCall::ObjectHandleOnStack wrapper);
+
+extern "C" void QCALLTYPE ComWrappers_RegisterIsRootedCallback();
+
+extern "C" CLR_BOOL QCALLTYPE TrackerObjectManager_HasReferenceTrackerManager();
+
+extern "C" CLR_BOOL QCALLTYPE TrackerObjectManager_TryRegisterReferenceTrackerManager(_In_ void* manager);
+
+extern "C" void QCALLTYPE TrackerObjectManager_RegisterNativeObjectWrapperCache(_In_ QCall::ObjectHandleOnStack cache);
+
+extern "C" CLR_BOOL QCALLTYPE TrackerObjectManager_IsGlobalPeggingEnabled();
+
+class GlobalComWrappersForMarshalling
+{
+public:
+    static bool TryGetOrCreateComInterfaceForObject(
+        _In_ OBJECTREF instance,
+        _Outptr_ void** wrapperRaw);
+
+    static bool TryGetOrCreateObjectForComInstance(
+        _In_ IUnknown* externalComObject,
+        _In_ INT32 objFromComIPFlags,
+        _Out_ OBJECTREF* objRef);
+};
+
+// Define "manually managed" definitions of the ComWrappers types
+// that are used in diagnostics and during GC.
+class GCHandleSetObject;
+class GCHandleSetEntryObject;
+
+class ManagedObjectWrapperHolderObject : public Object
+{
+    friend class CoreLibBinder;
+private:
+    OBJECTREF _releaser;
+    OBJECTREF _wrappedObject;
+public:
+    DPTR(InteropLib::ABI::ManagedObjectWrapperLayout) ManagedObjectWrapper;
+};
+
+class NativeObjectWrapperObject : public Object
+{
+    friend class CoreLibBinder;
+    OBJECTREF _comWrappers;
+    TADDR _externalComObject;
+    TADDR _inner;
+    CLR_BOOL _aggregatedManagedObjectWrapper;
+    CLR_BOOL _uniqueInstance;
+    OBJECTHANDLE _proxyHandle;
+    OBJECTHANDLE _proxyHandleTrackingResurrection;
+public:
+    OBJECTHANDLE GetProxyHandle() const
+    {
+        return _proxyHandle;
+    }
+
+    TADDR GetExternalComObject() const
+    {
+        return _externalComObject;
+    }
+};
+
+class ReferenceTrackerNativeObjectWrapperObject final : public NativeObjectWrapperObject
+{
+    friend class CoreLibBinder;
+    TADDR _trackerObject;
+    TADDR _contextToken;
+    int _trackerObjectDisconnected;
+    CLR_BOOL _releaseTrackerObject;
+    OBJECTHANDLE _nativeObjectWrapperWeakHandle;
+public:
+    TADDR GetTrackerObject() const
+    {
+        return (_trackerObject == (TADDR)nullptr || _trackerObjectDisconnected == 1) ? (TADDR)nullptr : _trackerObject;
```

Comment:
```suggestion
        return (_trackerObject == (TADDR)nullptr || _trackerObjectDisconnected) ? (TADDR)nullptr : _trackerObject;
```

A `CLR_BOOL` is allowed to be treated as a `bool`.